tests/phpunit/tests/reusable-blocks/test-integration.php

```diff
@@ -46,6 +49,93 @@ public function it_should_convert_reusable_block() {
 		);
 	}
 
+	/**
+	 * @test
+	 * @group wpmlcore-7651
+	 */
+	public function itShouldNotReRenderIfNotAReusableBlock() {
+		$blockContent = 'The original block content';
+		$block        = [ 'foo' => 'bar' ];
+
+		$translation = $this->getTranslation();
+		$translation->expects( $this->never() )->method( 'convertBlock' );
+
+		\WP_Mock::userFunction( 'render_block' )->never();
+
+		$subject = $this->getSubject( $translation );
+
+		$this->assertEquals(
+			$blockContent,
+			$subject->reRenderInnerReusableBlock( $blockContent, $block )
+		);
+	}
+
+	/**
+	 * @test
+	 * @group wpmlcore-7651
+	 */
+	public function itShouldNotReRenderIfBlockAlreadyConverted() {
+		$id           = 123;
+		$blockContent = 'The original block content';
+		$block        = [
+			'attrs' => [
+				'ref' => $id,
+			],
+		];
+
+		$translation = $this->getTranslation();
+		$translation->method( 'convertBlock' )
+			->with( $block )
+			->willReturnArgument( 0 );
+
+		\WP_Mock::userFunction( 'render_block' )->never();
+
+		$subject = $this->getSubject( $translation );
+
+		$this->assertEquals(
+			$blockContent,
+			$subject->reRenderInnerReusableBlock( $blockContent, $block )
+		);
+	}
+
+	/**
+	 * @test
+	 * @group wpmlcore-7651
+	 */
+	public function itShouldReRenderBlock() {
+		$id                = 123;
+		$convertedId       = 456;
+		$blockContent      = 'The original block content';
+		$translatedContent = 'The translated content';
+
+		$getBlock = function( $id ) {
+			return [
+				'attrs' => [
+					'ref' => $id,
+				],
+			];
+		};
+
+		$block          = $getBlock( $id );
+		$convertedBlock = $getBlock( $convertedId );
+
+		$translation = $this->getTranslation();
+		$translation->method( 'convertBlock' )
+			->with( $block )
+			->willReturn( $convertedBlock );
+
+		\WP_Mock::userFunction( 'render_block' )
+	        ->with( $convertedBlock )
```

Comment:
@strategio oops ... missing a bit of alignment here.

Reply:
Fixed :)